4MD/src/main/java/com/infoshareacademy/four_md/controller/BookCookController.java

```diff
@@ -1,32 +1,28 @@
 package com.infoshareacademy.four_md.controller;
 
+import com.infoshareacademy.four_md.model.entitiy.RecipeEntity;
+import com.infoshareacademy.four_md.service.jpaRepos.SearchEngineRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
+import java.util.List;
+
 @RestController
 @RequestMapping("/BookCook")
+@RequiredArgsConstructor
 public class BookCookController {
-//    private RecipeProvider recipeProvider;
-//
-//    public BookCookController(RecipeProvider recipeProvider) {
-//        this.recipeProvider = recipeProvider;
-//
-//    }
-//
-//    @GetMapping("/{id}")
-//    public Recipe getRecipe(@PathVariable int id) throws IOException{
-//            return recipeProvider.get(id);
-//    }
-//    @GetMapping("/all")
-//    public List<Recipe> getRecipes() throws IOException{
-//            return recipeProvider.getAll();
-//    }
-//    @PostMapping()
-//    public void addRecipe(@RequestBody Recipe recipe) throws IOException{
-//        recipeProvider.save(recipe);
-//    }
-//    @DeleteMapping("/{id}")
-//    public void deleteRecipe(@PathVariable int id) throws IOException{
-//        recipeProvider.remove(id);
-//    }
+
+    private final SearchEngineRepository searchEngineRepository;
+
+    @RequestMapping("/recipes/search/{name}")
+    public List<RecipeEntity> findByNameContains(@PathVariable String name) {
+        return searchEngineRepository.findByNameContains(name);
+    }
+
+    @RequestMapping(value = "/recipes/search/calories/", params = {"calMin", "calMax"})
+    public List<RecipeEntity> findByCaloriesBetween(@PathVariable int calMin, @PathVariable int calMax) {
```

Comment:
Ja mam jedno proste pytanie: JAKIM PRAWEM TAKI KOD TRAFIŁ NA CODE REVIEW i już nie chodzi mi o to, że dałeś złą adnotację przy argumentach (ciężko powiedzieć, jakie byłyby poprawne z twoją koncepcją), ale gdybyś minimalnie przetestował tem kod, to byś zauważył, że ta metoda po prostu NIE DZIAŁA